### PR TITLE
CASSANDRA-15980 - Improved log messaging for socket connection/disconnection 

### PR DESCRIPTION
**Pull request by jonmeredith**
See [CASSANDRA-15980](https://issues.apache.org/jira/browse/CASSANDRA-15980#)

### CODE REVIEW

File: src/java/org/apache/cassandra/net/InboundConnectionInitiator.java [hunk 1]
```diff
@@ -445,11 +454,11 @@ void setupMessagingPipeline(InetAddressAndPort from, int useMessagingVersion, in
             InboundMessageHandler handler =
                 settings.handlers.apply(from).createHandler(frameDecoder, initiate.type, pipeline.channel(), useMessagingVersion);
 
-            logger.info("{} connection established, version = {}, framing = {}, encryption = {}",
+            logger.info("{} messaging connection established, version = {}, framing = {}, encryption = {}",
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/net/InboundConnectionInitiator.java:457]: before

```
127.0.0.2:7000(127.0.0.1:65115)->127.0.0.1:7000-LARGE_MESSAGES-f53d678a connection established, version = 12, framing = CRC, encryption = enabled (openssl)
```

after

```
127.0.0.2:7000(127.0.0.1:65115)->127.0.0.1:7000-LARGE_MESSAGES-f53d678a messaging connection established, version = 12, framing = CRC, encryption = enabled (openssl)
```

Comment (dcapwell) [reply src/java/org/apache/cassandra/net/InboundConnectionInitiator.java:457]: personal preference, I prefer the old

File: src/java/org/apache/cassandra/net/SocketFactory.java [hunk 2]
```diff
@@ -237,6 +239,29 @@ static String encryptionLogStatement(EncryptionOptions options)
...
+    static String encryptionLogStatement(Channel channel, EncryptionOptions options)
+    {
+        if (options == null)
+            return "disabled";
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("enabled (factory=");
+        sb.append(SSLFactory.openSslIsAvailable() ? "openssl" : "jdk");
+
+        SslHandler sslHandler = channel.pipeline().get(SslHandler.class);
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/net/SocketFactory.java]: you pass null to channel but don't have any checks against it; in testing this patch found a case where this causes a NPE

```
ERROR [main] 2020-07-30 16:00:56,584 CassandraDaemon.java:800 - Exception encountered during startup
java.lang.NullPointerException: null
  at org.apache.cassandra.net.SocketFactory.encryptionLogStatement(SocketFactory.java:251)
  at org.apache.cassandra.net.InboundConnectionSettings.toString(InboundConnectionSettings.java:87)
  at java.base/java.lang.String.valueOf(String.java:2951)
  at java.base/java.lang.StringBuilder.append(StringBuilder.java:168)
  at org.apache.cassandra.net.InboundConnectionInitiator.bind(InboundConnectionInitiator.java:129)
```

File: src/java/org/apache/cassandra/net/InboundConnectionInitiator.java (hunk 1 above)
Comment (jonmeredith) [reply src/java/org/apache/cassandra/net/InboundConnectionInitiator.java:457]: It's to disambiguate between streaming connections so I'd prefer to keep it.

File: src/java/org/apache/cassandra/net/SocketFactory.java (hunk 2 above)
Comment (jonmeredith) [reply src/java/org/apache/cassandra/net/SocketFactory.java]: fixed

File: src/java/org/apache/cassandra/net/SocketFactory.java [hunk 3]
```diff
@@ -237,6 +240,30 @@ static String encryptionLogStatement(EncryptionOptions options)
...
+    static String encryptionLogStatement(Channel channel, EncryptionOptions options)
+    {
+        if (options == null)
+            return "disabled";
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("enabled (factory=");
+        sb.append(SSLFactory.openSslIsAvailable() ? "openssl" : "jdk");
+
+        final ChannelPipeline pipeline = channel.pipeline();
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/net/SocketFactory.java]: this doesn't fix the issue

```
public String toString()
     {
         return format("address: (%s), nic: %s, encryption: %s",
-                       bindAddress, FBUtilities.getNetworkInterface(bindAddress.address), SocketFactory.encryptionLogStatement(encryption));
+                       bindAddress, FBUtilities.getNetworkInterface(bindAddress.address), SocketFactory.encryptionLogStatement(null, encryption));
     }
```

The toString method now passes null as the channel, so the channel itself is null

Comment (jonmeredith) [reply src/java/org/apache/cassandra/net/SocketFactory.java]: Teach me to fix things at the end of the day.  Pushed up a fix. The silver lining is that it made me look a bit more carefully about the message and I think the current one is misleading so the fix improves the `toString() output to be more useful.

I've updated the JIRA with more details to get more visibility on it.

**Comment by jonmeredith** [comment]
Merged as c9b41c1f8ad03719918c4d3c29719056ae6b3995

**Pull request closed without merging**